java/test/jmri/jmrix/loconet/LnThrottleManagerTest.java

```diff
@@ -786,23 +780,17 @@ public void notifyDecisionRequired(LocoAddress address, DecisionType question) {
 
         // Wait for 3 messages, the set slot common and then the get slot refresh
         JUnitUtil.waitFor(() -> {
-            return !lnis.outbound.isEmpty();
+            return lnis.outbound.size() > 2;
         },"wait for message");
-        JUnitUtil.waitFor(() -> {
-            return !lnis.outbound.elementAt(lnis.outbound.size() -1).toString().equals("B5 09 10 00");
-        }, "COMMON");
-        JUnitUtil.waitFor(() -> {
-            return !lnis.outbound.elementAt(lnis.outbound.size() -1).toString().equals("BA 09 00 00");
-        }, "DISPATCH");
-        JUnitUtil.waitFor(() -> {
-            return !lnis.outbound.elementAt(lnis.outbound.size() -1).toString().equals("BB 09 00 00");
-        }, "Slot read");
+        
+        Assert.assertEquals("[B5 09 10 00, BA 09 00 00, B5 09 10 00]", lnis.outbound.toString());
```

Comment:
You might have degraded the test here, but I'm not sure. On line 789, the test previously expected four messages in the outbound queue. Now you only expect three messages. Before, the last expected message was `BB 09 00 00`. Now you expect `B5 09 10 00`. It seems to me that you should expect `[B5 09 10 00, BA 09 00 00, B5 09 10 00, BB 09 00 00]` and that line 789 should expect four messages. To make this work, you would need to change line 783 to wait for four messages.

Could you check this again? Thanks.

It would also be good to add some comments about the expected messages:

```
// B5 09 10 00 = COMMON
// BA 09 00 00 = DISPATCH
// B5 09 10 00 = COMMON
// BB 09 00 00 = Slot read
Assert.assertEquals("[B5 09 10 00, BA 09 00 00, B5 09 10 00]", lnis.outbound.toString());
```